examples/tic-tac-toe/cli/src/client.rs

```diff
@@ -0,0 +1,749 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use anyhow::{bail, Context, Result};
+use clap::Parser;
+use move_core_types::language_storage::StructTag;
+use shared_crypto::intent::Intent;
+use sui_keys::keystore::AccountKeystore;
+use sui_sdk::{
+    rpc_types::{
+        DevInspectArgs, DevInspectResults, DryRunTransactionBlockResponse, ObjectChange, SuiData,
+        SuiExecutionStatus, SuiObjectData, SuiObjectDataFilter, SuiObjectDataOptions,
+        SuiObjectResponse, SuiObjectResponseQuery, SuiProtocolConfigValue,
+        SuiTransactionBlockEffectsAPI, SuiTransactionBlockResponse,
+    },
+    wallet_context::WalletContext,
+    SuiClient,
+};
+use sui_types::{
+    base_types::{ObjectID, ObjectRef, SuiAddress},
+    crypto::PublicKey,
+    multisig::{MultiSig, MultiSigPublicKey},
+    object::Owner,
+    programmable_transaction_builder::ProgrammableTransactionBuilder,
+    signature::GenericSignature,
+    transaction::{
+        InputObjectKind, ObjectArg, ProgrammableTransaction, Transaction, TransactionData,
+        TransactionKind,
+    },
+    Identifier,
+};
+
+use crate::{
+    crypto::combine_keys,
+    game::{self, Game, GameKind, Winner},
+    turn_cap::TurnCap,
+};
+
+#[derive(Parser, Debug)]
+pub struct Connection {
+    /// The Sui CLI config directory, (default: ~/.sui/sui_config/client.yaml)
+    #[clap(long)]
+    config: Option<PathBuf>,
+
+    /// Object ID of the game's package.
+    #[clap(long, short, env = "PKG")]
+    package_id: ObjectID,
+}
+
+pub(crate) struct Client {
+    wallet: WalletContext,
+    package: ObjectID,
+}
+
+impl Client {
+    /// Create a new client that derives its active address and RPC from the CLI's config (found at
+    /// path `config`), and that expects to interact with the tic-tac-toe package at address
+    /// `package`.
+    pub(crate) fn new(conn: Connection) -> Result<Self> {
+        let Some(config) = conn.config.or_else(|| {
+            let mut default = dirs::home_dir()?;
+            default.extend([".sui", "sui_config", "client.yaml"]);
```

Comment:
Maybe for examples is ok to try to keep deps small, but probably `sui_config` should be used here to get the default dir.

Reply:
At least today, I interpret `sui_config` as an internal crate -- not intended to be used by people building stuff outside of our own binaries, so I wouldn't expect or encourage third parties to depend on it (separately from keeping things small, for demo purposes).